### PR TITLE
preserve tree-like structure we can add "what invalidated" logging

### PR DESCRIPTION
**Pull request by stefanpenner**
- [x] make it work
- [x] tests

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -72,30 +73,35 @@ proto.read = function (readTree) {
...
-      for (var i = 0, l = inputPaths.length; i < l; i++) {
-        dir = inputPaths[i];
-        keys = self.keysForTree(dir);
-        inputTreeHashes[i] = helpers.hashStrings(keys);
-
-        if (self._inputTreeCacheHash[i] !== inputTreeHashes[i]) {
-          invalidateCache = true;
-        }
-      }
+  .then(function(inputPaths) {
```
Comment (rwjblue) [inline index.js]: Can you fix the indentation here?

Basically it should be something like the following:

```javascript
return mapSeries(this.inputTrees, readTree)
  .then(function(inputPaths) {
    /* stuff here */
  });
```

File: tests/index.js [hunk 2]
```diff
@@ -67,7 +69,7 @@ describe('broccoli-caching-writer', function(){
     });
   });
 
-  describe('write', function() {
+  describe.only('write', function() {
```
Comment (rwjblue) [inline tests/index.js]: `.only`

**Comment by stefanpenner** [comment]
@rwjblue updated according to your feedback

**Comment by rwjblue** [comment]
Looks good.  I'll merge and release, but I would like to have a followup PR that actually tests `keysForTree` and the `Key.prototype.equals` method in more detail.  I'll add an issue so I remember to look at doing that later...

**Comment by rwjblue** [comment]
released in 0.5.5.

**Comment by timmfin** [comment]
Nice :+1:

**Pull request merged**